pkg/platform/abstract/platform.go

```diff
@@ -1691,6 +1695,31 @@ func (ap *Platform) validateTriggers(functionConfig *functionconfig.Config) erro
 	return nil
 }
 
+func (ap *Platform) validateBatchConfiguration(functionConfig *functionconfig.Config) error {
+
+	for _, triggerInstance := range functionConfig.Spec.Triggers {
+		if triggerInstance.Batch == nil {
+			continue
+		}
+
+		if functionconfig.BatchModeEnabled(triggerInstance.Batch) &&
+			(!functionconfig.TriggerKindSupportsBatching(triggerInstance.Kind) ||
+				!functionconfig.RuntimeSupportsBatching(functionConfig.Spec.Runtime)) {
+			ap.Logger.WarnWith("Batching is not supported for given runtime and kind - batching configuration is ignored",
+				"runtime", functionConfig.Spec.Runtime,
+				"triggerKind", triggerInstance.Kind)
+		}
```

Comment:
See [this comment ](https://github.com/nuclio/nuclio/pull/3244#discussion_r1612063454)- splitting the checks to `trigger` and `runtime` separately results with a clearer error message.
When it's together - how can the user know which doesn't support batching? runtime or tigger?